test/testdata/rewriter/chained_sig.rb

```diff
@@ -0,0 +1,182 @@
+  def blah; end
+
+  sig.on_failure(:soft, notify: "me")
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Cannot use `on_failure` outside of a sig block
+    # ^^^^^^^^^^ error: Method `on_failure` does not exist on `NilClass`
+  def bip; end
+
+  sig.type_parameters(:T)
+# ^^^^^^^^^^^^^^^^^^^^^^^ error: Cannot use `type_parameters` outside of a sig block
+    # ^^^^^^^^^^^^^^^ error: Method `type_parameters` does not exist on `NilClass`
+  def bop; end
+end
+
+module Interface
+  extend T::Sig
+  extend T::Helpers
+
+  interface!
+
+  sig.abstract { void }
+  def foo; end
+end
+
+class DuplicateBlock
+  extend T::Sig
+  include Interface
+
+  sig.override { void }.final { void }
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Cannot add more signature statements after the declaration block
+  def foo; end
+end
+
+class WithoutRuntime
+  extend T::Sig
+  include Interface
+
+  T::Sig::WithoutRuntime.sig.final { void }
+  def baz; end
+
+  T::Sig::WithoutRuntime.sig.params(a: Integer)
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Cannot use `params` outside of a sig block
+                           # ^^^^^^ error: Method `params` does not exist on `NilClass`
+  def bar(a); end
+
+  T::Sig::WithoutRuntime.sig.override { void }.final { void }
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Cannot add more signature statements after the declaration block
+  def foo; end
+end
+
+class MissingBlocks
+  extend T::Sig
+
+  sig
+# ^^^ error: Signature declarations expect a block
+  def foo; end
+
+  sig.final
+# ^^^^^^^^^ error: Signature declarations expect a block
+    # ^^^^^ error: Method `final` does not exist on `NilClass`
+  def bar; end
+end
+
+class InstructionSequenceInSig
+  extend T::Sig
+
+  sig.final { Kernel.puts ""; void }
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Malformed signature: cannot have multiple instructions inside a signature block
+    # ^^^^^ error: Method `final` does not exist on `NilClass`
+                            # ^^^^ error: Method `void` does not exist on `T.class_of(InstructionSequenceInSig)`
+  def foo; end
+end
+
+class DuplicateInvocations
+  extend T::Sig
+  extend T::Helpers
+  include Interface
+
+  abstract!
+
+  sig.overridable { overridable.void }
+                  # ^^^^^^^^^^^ error: Duplicate invocation of `overridable` in signature declaration
+  def bar; end
+
+  sig.override { override.void }
+               # ^^^^^^^^ error: Duplicate invocation of `override` in signature declaration
+  def foo; end
+
+  sig.abstract { abstract.void }
+               # ^^^^^^^^ error: Duplicate invocation of `abstract` in signature declaration
+  def baz; end
+end
+
+class DuplicateButMissingBlock
+  extend T::Sig
+  include Interface
+
+  sig.final { void }.override
+                   # ^^^^^^^^ error: Method `override` does not exist on `NilClass`
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Signature declarations expect a block
```

Comment:
This second error seems not great, since we *are* providing a block, it's just that the entire declaration here is not in the format that we expect.  Can we do better than this?

Reply:
I'm not sure. We only keep track of the previous `sig` send and not `sig.final`.

What do you think of changing the loc to `funLoc` instead and underlying `override`. This will also affect cases like `sig.final`